src/test/java/spoon/test/reference/TypeReferenceTest.java

```diff
@@ -166,11 +158,36 @@ public void testNullReferenceSubtype() throws Exception {
 		assertFalse(nullRef.isSubtypeOf(ref));
 
 	}
-	
-    @Test
-    public void unboxTest() {
-        Factory factory = new Launcher().createFactory();
-        CtTypeReference<Boolean> boxedBoolean = factory.Class().createReference(Boolean.class);
-        assertEquals(boxedBoolean.unbox().getActualClass(), boolean.class);
-    }
+
+	@Test
+	public void unboxTest() {
+		Factory factory = new Launcher().createFactory();
+		CtTypeReference<Boolean> boxedBoolean = factory.Class().createReference(Boolean.class);
+		assertEquals(boxedBoolean.unbox().getActualClass(), boolean.class);
+	}
+
+	@Test
+	public void testToStringEqualityBetweenTwoGenericTypeDifferent() throws Exception {
+		final Launcher launcher = new Launcher();
```

Comment:
contract: generic type references with different bounds should not be considered equals